tool/update_github_emojis.dart

```diff
@@ -0,0 +1,338 @@
+/// { 'shortcode':'emojifilename' ... }
+/// The filenames are simply a list of all of the hex string of the
+/// *essential* unicode codepoints representing the emoji.
+/// These sequences exclude the unicode join zero width (0x200D) and
+/// variation select (0xFE0F) modifiers. (We will need to add these in to
+/// build our actually unicode strings representing the emoji).
+/// Multiple unicode codepoints are separated by '-'.
+/// Examples filenames (single and double code point examples):
+///  - "https://github.githubassets.com/images/icons/emoji/unicode/1f643.png?v8"
+///  - "https://github.githubassets.com/images/icons/emoji/unicode/1f1fa-1f1fe.png?v8"
+///  - "https://github.githubassets.com/images/icons/emoji/unicode/1f469-1f469-1f467-1f466.png?v8"
+/// NOTE: Some filenames will be github 'custom' emojis that have no unicode
+/// equivalent and these will not have hex codepoints, only the github custom name.
+/// We will ingore these (there are only a 19 and they are mostly pixel art from
+/// the old Doom game).
+/// Example github custom emoji filename:
+///  - "https://github.githubassets.com/images/icons/emoji/godmode.png?v8",
+String parseGithubFilenameIntoUnicodeString(String emojiFilename) {
+  const variationSelector = 0xFE0F;
+  const zeroWidthJoiner = 0x200D;
+
+  try {
+    final String? rawHexList =
+        parseGithubEmojiUnicodeFromFilename.firstMatch(emojiFilename)?.group(1);
+    if (rawHexList == null) {
+      // This is a github custom emoji and it is represent by PNG only and
+      // there is no equivalent unicode.  We have to ingore.
+      return '';
+    }
+    bool legacyUsedVariationCode = false;
+    if (legacyEmojisUsedVariationModifier.contains(rawHexList)) {
+      legacyUsedVariationCode = true;
+    }
+    final List<int> rawCodePointsHex = rawHexList
+        .split('-')
+        .map((hexstr) => int.parse(hexstr, radix: 16))
+        .toList();
+    final List<int> codePointsHex = [];
+
+    if (legacyUsedVariationCode) {
+      // just add single variation selector
+      codePointsHex.addAll(rawCodePointsHex);
+      codePointsHex.add(variationSelector);
+    } else {
+      // now insert the join zero width and variation select modifying unicode chars
+      for (var i = 0; i < rawCodePointsHex.length; i++) {
+        final codePointAtIndex = rawCodePointsHex[i];
+        codePointsHex.add(codePointAtIndex);
+        if (i < (rawCodePointsHex.length - 1)) {
+          codePointsHex.add(variationSelector);
+          // # and 0-9 don't use zero width joiners.
+          if (codePointAtIndex == 0x23 ||
+              (codePointAtIndex >= 0x30 && codePointAtIndex <= 0x39)) {
+            // no zerowidth joiner
+          } else {
+            codePointsHex.add(zeroWidthJoiner);
+          }
+        }
+      }
+    }
+    return String.fromCharCodes(codePointsHex);
+  } catch (e) {
+    print('Invalid/Non-Conformant emoji filename encountered $emojiFilename !');
```

Comment:
Let's indicate the interpolation with some delimiter, in case it ever gets blank or something: `encountered: "$emojiFilename"!');`.

Reply:
I don't think this has been addressed.